src/carma-streets/parse_ros_bags.py

```diff
@@ -0,0 +1,57 @@
+import rosbag
+import csv
+import pandas as pd
+import matplotlib.pyplot as plt
+import argparse
+from pathlib import Path
+
+
+
+def parse_ros_bag_file(ros_bag_file, csv_dir):
+    ros_bag_file_path = Path(ros_bag_file)
+    csv_dir_path = Path(csv_dir)
+
+# calculate the offset between ros ckock and the sim clock
+def get_time_offset(ros_bag_file):
+    sim_time_topic_name = '/sim_clock'
+    time_offset = 0
+    with rosbag.Bag(ros_bag_file, 'r') as bag:
+        for _, msg, t in bag.read_messages(topics=[sim_time_topic_name]):
+            time_offset = (t.secs * 1000 + t.nsecs // 1000000) - (msg.clock.secs * 1000 + msg.clock.nsecs // 1000000)
+            return time_offset
+
+
+
+def get_detected_objects(ros_bag_file, outputfile, time_offset):
+    detected_obj_topic_name = '/environment/fused_external_objects'
+    print(f'time_offset {time_offset}')
+    messages = []
+    with rosbag.Bag(ros_bag_file, 'r') as bag:
+        for topic, msg, _ in bag.read_messages(topics=[detected_obj_topic_name]):
+            messages.extend(msg.objects)
+
+    if outputfile.exists():
+        print(f'Output file {outputfile} already exists. Overwriting file.')
+    with open(outputfile, 'w', newline='') as file:
+        writer = csv.writer(file)
+        writer.writerow(['Timestamp (ms)', 'ObjID', 'Positionx',  'Positiony']) 
+        for message in messages:
+            timestamp_ms = message.header.stamp.secs * 1000 + message.header.stamp.nsecs // 1000000 - time_offset
+            writer.writerow([timestamp_ms, message.id % 1000, message.pose.pose.position.x, message.pose.pose.position.y]) 
+
+    print(f"Sorted data from topic '{detected_obj_topic_name}' saved to '{outputfile}'")
+    print(f"Calculated sim time offset: '{time_offset}'")
+
+def main():
+    parser = argparse.ArgumentParser(description='Script to parse ROS Bags into CSV data')
+    parser.add_argument('--ros-bag-file', help='Directory containing ROS Bags.', type=str, required=True) 
```

Comment:
The description says "directory," but the argument name has "file." We should be consistent.

Reply:
updated